crates/core/src/dht/dht_store.rs

```diff
@@ -297,12 +302,37 @@ impl DhtStore {
         &mut self.actions
     }
 
-    pub(crate) fn next_queued_holding_workflow(&self) -> Option<&PendingValidation> {
-        self.queued_holding_workflows.front()
+    pub(crate) fn next_queued_holding_workflow(
+        &self,
+    ) -> Option<(&PendingValidation, Option<Duration>)> {
+        for (pending, maybe_delay) in self.queued_holding_workflows.iter() {
+            if let Some((time_of_dispatch, delay)) = maybe_delay {
+                let maybe_time_elapsed = time_of_dispatch.elapsed();
+                if let Ok(time_elapsed) = maybe_time_elapsed {
+                    if time_elapsed < *delay {
+                        continue;
+                    }
+                }
+                return Some((pending, Some(*delay)));
+            }
+            return Some((pending, None));
+        }
+        None
     }
 
     pub(crate) fn has_queued_holding_workflow(&self, pending: &PendingValidation) -> bool {
-        self.queued_holding_workflows.contains(pending)
+        for (current, _) in self.queued_holding_workflows.iter() {
+            if current == pending {
+                return true;
+            }
+        }
+        false
```

Comment:
Could be a one liner

self.queued_holding_workflows.iter().any(|(current,_)|current==pending)